common/src/stack/command/stack/commands/load/json/__init__.py

```diff
@@ -0,0 +1,141 @@
+	"""
+	Load configuration data from the provided json document. If no arguments
+	are provided then all plugins will be run.
+
+	<param optional='0' type='string' name='file'>
+	The json document containing the configuration information.
+	</param>
+
+	<arg optional='1' type='string' name='software'>
+	Load pallet, cart, and box data. If the pallet exists on a remote
+	server that requires authentication, be sure to provide the username
+	and password in the corresponding json fields.
+	</arg>
+
+	<arg optional='1' type='string' name='host'>
+	Load name, rack, rank, interface, attr, firewall, box, appliance,
+	comment, metadata, environment, osaction, route, group, partition,
+	and controller data for each host.
+	</arg>
+
+	<arg optional='1' type='string' name='network'>
+	Load name, address, gateway, netmask, dsn, pxe, mtu, and zone
+	data for each network.
+	</arg>
+
+	<arg optional='1' type='string' name='global'>
+	Load attr, route, firewall, partition, and controller data for
+	the global scope.
+	</arg>
+
+	<arg optional='1' type='string' name='os'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each os.
+	</arg>
+
+	<arg optional='1' type='string' name='appliance'>
+	Load name, attr, route, firewall, partition, and controller
+	data for each appliance.
+	</arg>
+
+	<arg optional='1' type='string' name='group'>
+	Load name data for each group.
+	</arg>
+
+	<arg optional='1' type='string' name='bootaction'>
+	Load name, kernel, ramdisk, type, arg, and os data for each
+	bootaction.
+	</arg>
+	"""
+
+	def checks(self, args):
+
+		# Make sure that the user is not attepting to add more than one frontend
+		if ('host' in args or not args) and 'host' in self.data:
+			current_frontend = json.loads(self.command('list.host', [ 'a:frontend', 'output-format=json' ]))[0]['host']
+			for host in self.data['host']:
+				if host['appliance'] == 'frontend' and host['name'] != current_frontend:
+					raise CommandError(self, 'frontend name in the json document does not match the current frontend name')
+
+		# Make sure that all of the pallets at least have URLs
+		if ('software' in args or not args) and 'software' in self.data and 'pallet' in self.data['software']:
+			for pallet in self.data['software']['pallet']:
+				if pallet['url'] == None:
+					raise CommandError(self, f'pallet {pallet["name"]} {pallet["version"]} has no url')
+
+
+	def run(self, params, args):
+
+		filename, = self.fillParams([
+			('file', None, True),
+		])
+
+		if not os.path.exists(filename):
+			raise CommandError(self, f'file {filename} does not exist')
+
+		with open (os.path.abspath(filename), 'r') as f:
+			try:
+				self.data = json.load(f)
+			except ValueError:
+				raise CommandError(self, 'Invalid json document')
+
+		# run a few pre checks
+		self.checks(args)
+
+		# make a backup of the database in its current state in the event of any errors
+		s = subprocess.run(['mysqldump', 'cluster'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
As mentioned, replace this with a call to `/etc/cron.daily/backup-whatever`.  And check that the return code is zero.